pyhive/hive.py

```diff
@@ -207,6 +196,109 @@ def sasl_factory():
             self._transport.close()
             raise
 
+    @staticmethod
+    def create_http_transport(host, port, http_path, username, password,
+                              kerberos_service_name, auth):
+        if port is None:
+            port = 10001
+        if auth is None:
+            auth = 'NONE'
+        if http_path is None:
+            http_path = '/'
+
+        socket = TCookieHttpClient('http://{}:{}{}'.format(host, port, http_path))
```

Comment:
It would be great if we could also pass through a `http_protocol`, either `http` or `https` and have this set.